lib/src/dependency.dart

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:json_annotation/json_annotation.dart';
+import 'package:pub_semver/pub_semver.dart';
+
+abstract class Dependency {
+  Dependency._();
+
+  /// Returns `null` if the data could not be parsed.
```

Comment:
[nit] This seems like a detail, not a headline/summary

Also it seems like sometimes it throws instead of returning null

Reply:
The null value is used by the caller to throw since it has more knowledge. 

Move `_getDeps` to that file and made the constructor private - so it's more of an implementation detail.

https://github.com/dart-lang/pubspec_parse/pull/4/commits/3eaf3fbcb01d4ccbd63844c6359d812d91210244